### ISSUE TITLE
Okta.AspNetCore Authentication: OktaWebApi + Okta Api Token

### ISSUE THREAD

**Issue by vsspt** [okta/okta-aspnet#137]
I have a .net core 3.0 Rest API, secured with Okta.AspNetCore OktaWebApi : 

```
            services.AddAuthentication(options =>
                {
                    options.DefaultAuthenticateScheme = OktaDefaults.ApiAuthenticationScheme;
                    options.DefaultChallengeScheme = OktaDefaults.ApiAuthenticationScheme;
                    options.DefaultSignInScheme = OktaDefaults.ApiAuthenticationScheme;
                })
                .AddOktaWebApi(new OktaWebApiOptions
                {
                    OktaDomain = Environment.GetEnvironmentVariable("OKTA_DOMAIN"),
                });
```

This works perfectly!

I want to add Authentication using Okta API Tokens: 
- https://developer.okta.com/docs/guides/create-an-api-token/overview

Is there a way to use Okta.AspNetCore to enable multiple Authentication schemas, as OktaWebApi + Okta API Token?

Thank you!

**Comment by bryanapellanes-okta** [comment]
@vsspt ,
Thank you for submitting this issue.  We will review internally to determine the appropriate approach to achieve what you're requesting.  We'll respond here when more information is available.  Stay tuned!

Thanks!
Bryan

**Comment by vsspt** [comment]
Thank you very much for such a fast response.

**Comment by laura-rodriguez** [comment]
Hi @vsspt ,

Can you provide us more details of your use case here? 

When using an access token (bearer token) obtained via OIDC we can validate it at the app level, but we cannot do that with the API token. When you call any Okta endpoints with an invalid API token, the Okta API will return the corresponding error.

**Comment by vsspt** [comment]
Hi @laura-rodriguez, thank you for your response.

We have an Okta server instance that acts as the Identity provider to users Stored in Okta, and also for users from external Idps (Azure, etc).  

We have several Micro-services, developed in .net core 3.x, secured with Okta.AspNetCore OktaWebApi, with all Endpoints authorized.

Currently, we have the following Use Case in place:
 - As a User, after authenticating successfully in the System, an access token (bearer token) is generated, which will be validated by Okta.AspNetCore:

Startup.cs 
```
            services.AddAuthentication(options =>
                {
                    options.DefaultAuthenticateScheme = OktaDefaults.ApiAuthenticationScheme;
                    options.DefaultChallengeScheme = OktaDefaults.ApiAuthenticationScheme;
                    options.DefaultSignInScheme = OktaDefaults.ApiAuthenticationScheme;
                })
                .AddOktaWebApi(new OktaWebApiOptions
                {
                    OktaDomain = Environment.GetEnvironmentVariable("OKTA_DOMAIN"),
                });
```

Controller.cs

```
        [HttpGet]
        [Authorize]
        public async Task<ActionResult<Something>> GetSomething()
```

As mentioned, everything is working perfectly.

We now want to implement a second Use-Case:
- As an external Component, i want to be able to authenticate without having a User/User credentials, but instead, use Okta API tokens.

The goal here, is to provide an ID Token to certain external components, that would need to access our micro-services.
Basically, we want to be able to support, at the same time, authentication with an access token (bearer token) or an Okta API Token.

Ideally, this would be done at the Okta.AspNetCore OktaWebApi level.

If this is not possible, there are several ways to do it, such us:
- Creating a .net core Custom Middleware,
- Create a new AuthenticationHandler
- etc

Nevertheless, i would like to do this in a seamless way, and if possible, following the same patterns use by Okta.AspNetCore OktaWebApi.

I hope this clarifies what i want to achieve, and if not, please contact me for further information.

Regards,
Vasco.

**Comment by laura-rodriguez** [comment]
Hi Vasco,

Thanks for the details.

API tokens are not used for authentication and we do not recommend that approach. API Tokens are highly privileged tokens, so it is not a good idea to pass them around.

That being said, if what you want is to implement a client credentials flow for a machine-to-machine scenario, then I suggest looking into [OAuth for Okta APIs](https://github.com/okta/okta-sdk-dotnet#oauth-20). 

I hope this helps.

**Comment by vsspt** [comment]
Thank you for you response.

I will need a few days to research, please do not close this issue, i'll post as soon as i have new informations.

Thank you very much!

**Comment by laura-rodriguez** [comment]
Closing due to inactivity. Feel free to reopen if it's necessary.
